File: sqlite3/vb/read/sqlite3_read.vb
' ----------------------------------------------------------------
'
'	read/sqlite3_read.vb
'
'					Aug/25/2015
'
'
' ----------------------------------------------------------------
imports System
imports Mono.Data.Sqlite

' ----------------------------------------------------------------
Module sqlite3_read

' ----------------------------------------------------------------
Sub Main ()
	Dim clArgs() As String = Environment.GetCommandLineArgs()
	console.writeLine("*** 開始 ***")

	Dim file_sqlite3 as String = clArgs(1)

	Dim str_connect as String = "Data Source = " + file_sqlite3 _
			+ ";UTF8Encoding=True;Version=3"


	Dim conn as SqliteConnection = new SqliteConnection (str_connect)

	conn.Open()

	Dim str_sql as String = "select id,name,population,date_mod from cities"
	Dim command as SqliteCommand = new SqliteCommand(str_sql, conn)

	Dim dr as SqliteDataReader = command.ExecuteReader()

	Dim delim as String = vbTab

	While dr.Read()
		For it As Integer = 0 To (dr.FieldCount - 1) 
			console.write (dr.GetValue(it).ToString ())
			console.write (delim)
		Next
		console.writeLine ()

	End While

	conn.Close()

	console.writeLine("*** 終了 ***")
End Sub
' ----------------------------------------------------------------
End Module
' ----------------------------------------------------------------

File: sqlite3/csharp/read/sqlite3_read.cs
// -------------------------------------------------------------------
/*
	sqlite3_read.cs


					Apr/07/2015
*/
// -------------------------------------------------------------------
using	System;
using	Mono.Data.Sqlite;

// -------------------------------------------------------------------
class sqlite3_read
{
/* ------------------------------------------------------------------- */
static int Main (string[] args)
{
	string out_str;

	Console.Error.WriteLine ("*** 開始 ***");

	if (args.Length < 1)
		{
		Console.WriteLine ("*** error ***");
		return 1;
		}

	string file_sqlite3 = args[0];

	string str_connect = @"Data Source = " + file_sqlite3
			+ ";UTF8Encoding=True;Version=3";
 
	SqliteConnection conn = new SqliteConnection (str_connect);

	string str_query =
		"select ID,Name,Population,date_mod from cities";

	SqliteCommand cmd = new SqliteCommand (str_query,conn);

	conn.Open();
  
	SqliteDataReader dr = cmd.ExecuteReader();

	while	(dr.Read())
		{
		out_str = dr["ID"].ToString ()
			+ "\t" + dr["NAME"].ToString()
			 + "\t" +  dr["POPULATION"].ToString()
			 + "\t" +  dr["DATE_MOD"].ToString();

		Console.WriteLine (out_str);
		}
  
	dr.Close();
	dr.Dispose();
	cmd.Dispose();
	conn.Close();
	conn.Dispose();

	Console.Error.WriteLine ("*** 終了 ***");

	return	0;
}
// -------------------------------------------------------------------
}

// -------------------------------------------------------------------
